File: src/Department.java


import java.util.ArrayList;

public class Department {    // Department class
    private final DepartmentID _departmentID;
    private ArrayList<Lecturer> _lecturers;
    private FacultyID _facultyID;

    public Department(DepartmentID departmentID, FacultyID facultyID) {                      // constructor
        _departmentID = departmentID;
        _facultyID = facultyID;
    }

    public boolean addLecturer(Lecturer lecturer) {           // add lecturer method
        _lecturers.add(lecturer);
        return true;
    }
    public DepartmentID getDepartmentID() {                             // get department ID method
        return _departmentID;
    }
    public boolean removeLecturer(Lecturer lecturer,User user) {        // remove lecturer method
        _lecturers.remove(lecturer);
        return true;
    }

    public FacultyID get_facultyID() {
        return _facultyID;
    }

}
class DepartmentID {                                    // DepartmentID class
    private final int _departmentID;
    private final String _departmentName;
    public DepartmentID(int departmentID, String departmentName) {    // constructor
        _departmentID = departmentID;
        _departmentName = departmentName;
    }
    public String getDepartmentName() {                     //department name getter
        return _departmentName;
    }
    public int getDepartmentID() {                     //department ID getter
        return _departmentID;
    }
}

File: pythonCode/Department.py
import Faculty
import logging

from pythonCode.Logger import setup_logger
logger = setup_logger("Department")

class Department:
    def __init__(self, departmentID, facultyID):
        logging.getLogger().handlers.clear()
        logger.info(f"{self.__class__.__name__} classes created.")
        self._departmentID = departmentID
        self._facultyID = facultyID
        self._lecturers = list()

    @staticmethod
    def from_dict(data):
        return Department(DepartmentID.from_dict(data["_departmentID"]), Faculty.FacultyID.from_dict(data["_facultyID"]) )


    def addLecturer(self, lecturer):
        logger.info("Adding lecturer to the department.")
        self._lecturers.append(lecturer)
        return True

    def getDepartmentID(self):
        logger.info(f"The getDepartmentID method in the {self.__class__.__name__} class is called.")
        return self._departmentID

    def get_facultyID(self):
        logger.info(f"The get_facultyID method in the {self.__class__.__name__} class is called.")
        return self._facultyID


class DepartmentID:
    def __init__(self, departmentID, departmentName):
        logging.getLogger().handlers.clear()
        logger.info(f"{self.__class__.__name__} classes created.")
        self._departmentID = departmentID
        self._departmentName = departmentName

    @staticmethod
    def from_dict(data):
        return  DepartmentID(data["_departmentID"], data["_departmentName"])

    def getDepartmentID(self):
        logger.info(f"The getDepartmentID method in the {self.__class__.__name__} class is called.")
        return self._departmentID

    def getDepartmentName(self):
        logger.info(f"The getDepartmentName method in the {self.__class__.__name__} class is called.")
        return self._departmentName
